src/libultra/voice/voicecontrolgain.c

```diff
@@ -1,5 +1,41 @@
-#include "global.h"
+/**
+ * File: voicecontrolgain.c
+ *
+ * Adjusts the input gain of the Voice Recognition System
+ */
 
-u8 D_80098180[] = { 0x00, 0x80, 0x40, 0xC0, 0x20, 0xA0, 0x60, 0xE0 };
+#include "ultra64/controller_voice.h"
+#include "ultra64/os_voice.h"
+#include "io/controller.h"
+#include "macros.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/voicecontrolgain/osVoiceControlGain.s")
+u8 sDigitalTable[] = { 0x00, 0x80, 0x40, 0xC0, 0x20, 0xA0, 0x60, 0xE0 };
+
+s32 osVoiceControlGain(OSVoiceHandle* hd, s32 analog, s32 digital) {
+    s32 errorCode;
+    u8 cmd;
+
+    if (analog == 0) {
+        cmd = 0x18;
+    } else {
+        cmd = 0x98;
```

Comment:
Do these have defines in controller.h? If not they probably should.

Reply:
https://ultra64.ca/files/documentation/online-manuals/man/pro-man/pro26/26-08.html#osVoiceControlGain()
![image](https://github.com/zeldaret/mm/assets/47598039/c2929062-f7ff-4297-b83b-b43d5dd2bf59)

Any suggestions? Would these command names be known or would we have to pick a reasonable name ourselves?

Reply:
Similarly for the digital: 
`cmd = sDigitalTable[digital] + 2;`
`u8 sDigitalTable[] = { 0x00, 0x80, 0x40, 0xC0, 0x20, 0xA0, 0x60, 0xE0 };`
(I wonder about the +2)
![image](https://github.com/zeldaret/mm/assets/47598039/b6036f30-401a-4f34-b78b-80f1420f6779)

Reply:
Bump on this. Any libultra folks have a name suggestions? Can always leave it to later as well

Reply:
Fwiw, ultralib has just left them as hex for the time being, though defines sound fine for now (though that +2 is still weird).
Would probably also want to do the same for the `sCmds` array in `voiceinit`.